Handle conflicting experiments when opening

tahini opened the pull request:
When opening a trace with the same name as another experiment, the experiment
that contains it will have a conflict on the server. The client should
retry posting it with a different name until it succeeds or another
error occurs.

The title of the trace viewer should also be updated to represent the
actual name of the experiment.

Signed-off-by: Geneviève Bastien <gbastien+lttng@versatic.net>

tahini commented:
This goes with Trace server commit: https://git.eclipse.org/r/#/c/163272/

tahini commented on viewer-prototype/src/common/experiment-manager.ts:
```diff
@@ -87,11 +88,32 @@ export class ExperimentManager {
+            let conflictResolutionResponse = experimentResponse;
+            let i = 1;
+            while (conflictResolutionResponse.getStatusCode() === 409) {
+                conflictResolutionResponse = await handleConflict(this.tspClient, i);
```
forgot i++

bhufmann replied:
Ahh I missed that. I only tried having one conflict. I'm waiting for an update.

tahini commented:
I updated the patch, and also added the management of the trace conflict, even though there is no such conflicts anymore from the trace server, but if we add some, it will be handled here.
